### PR TITLE
refactor endpoint slice to avoid unnecessary redirections

### PR DESCRIPTION
**Pull request by ramaraochavali**
We have lot of redirections like onEvent->handleEvent->processEvent making the code hard to follow. All of this code was written like this to support both endpoint and endpoint slices. Now that endpoints have been removed, we do not need this unnecessary redirections and abstractions

There is no functional change

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Dual Stack
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

**Review by hzxuzhonghu — COMMENTED** [review]
WIth the new krt, i think these code will be updated. So it seems we do not need to modify temporarily

**Comment by ramaraochavali** [comment]
@hzxuzhonghu  It will be updated with krt, but it will simplify the adoption of krt + there are other changes that are not related to events.
